graspologic/layouts/auto.py

```diff
@@ -0,0 +1,347 @@
+    precisely ``max_edges``.
+
+    In addition to pruning edges by weight, this function also only operates over the
+    largest connected component in the graph.
+
+    After dimensionality reduction, sizes are generated for each node based upon
+    their degree centrality, and these sizes and positions are further refined by an
+    overlap removal phase. Lastly, a global partitioning algorithm
+    (:func:`graspologic.partition.leiden`) is executed for the largest connected
+    component and the partition ID is included with each node position.
+
+    Parameters
+    ----------
+    graph : :class:`networkx.Graph`
+        The graph to generate a layout for. This graph may have edges pruned if the
+        count is too high and only the largest connected component will be used to
+        automatically generate a layout.
+    min_dist : float
+        The effective minimum distance between embedded points. Default is ``0.75``.
+        Smaller values will result in a more clustered/clumped embedding where nearby
+        points on the manifold are drawn closer together, while larger values will
+        result on a more even dispersal of points. The value should be set relative to
+        the ``spread`` value, which determines the scale at which embedded points will
+        be spread out.
+    n_neighbors : int
+        The size of local neighborhood (in terms of number of neighboring sample points)
+        used for manifold approximation. Default is ``25``. Larger values result in
+        more global views of the manifold, while smaller values result in more local
+        data being preserved.
+    max_edges : int
+        The maximum number of edges to use when generating the embedding.  Default is
+        ``10000000``. The edges with the lowest weights will be pruned until at most
+        ``max_edges`` exist. Warning: this pruning is approximate and more edges than
+        are necessary may be pruned. Running in 32 bit enviornment you will most
+        likely need to reduce this number or you will out of memory.
+
+    Returns
+    -------
+    Tuple[nx.Graph, List[NodePosition]]
+        The largest connected component and a list of NodePositions for each node in
+        the largest connected component. The NodePosition object contains:
+        - node_id
+        - x coordinate
+        - y coordinate
+        - size
+        - community
+
+    References
+    ----------
+    .. [1] McInnes, L, Healy, J, UMAP: Uniform Manifold Approximation and Projection
+        for Dimension Reduction, ArXiv e-prints 1802.03426, 2018
+    .. [2] Böhm, Jan Niklas; Berens, Philipp; Kobak, Dmitry. A Unifying Perspective
+        on Neighbor Embeddings along the Attraction-Repulsion Spectrum. ArXiv e-prints
+        2007.08902v1, 17 Jul 2020.
+    """
+
+    lcc_graph, tensors, labels = _node2vec_for_layout(graph, max_edges)
+    points = umap.UMAP(min_dist=min_dist, n_neighbors=n_neighbors).fit_transform(
+        tensors
+    )
```

Comment:
I don't see this dependency being added to the package anywhere am I missing it somewhere?